Setting_up_Travis_CI_and_Coveralls.md

```diff
@@ -0,0 +1,107 @@
+2. Click the [+] button on the left next to the "My Repositories" header. This will bring up a list of all repositories that you can activate.
+3. Activate that repo by clicking the gray toggle-slider button. That's it! You can modify settings on particular repos by clicking the gear button next to the activation toggle. By default, a Travis-CI job is run on every pull request and every commit.
+4. Add a build badge to the README.md in the root of your repo. Clicking on the repository under the "My Repository" heading will take you to a build status page. In the header of this page is a little badge, as circled in the image below.
+
+![Travis-CI badge example](images/travis-ci-badge.png)
+
+Click on that image to get a popup with several options for the link style (in the pulldown menu on the left). Select Markdown, then you can copy and paste that link into your README.md file.
+
+## <a name="travis-config"></a>Travis-CI script configuration
+
+Triggering the automated tests requires a little configuration in your repo. This means creating, in the root of your repo, a YAML-formatted file called ```.travis.yml```. Travis-CI has some very detailed documentation available on how to create these here: http://docs.travis-ci.com/ 
+
+The remainder of this section will include examples in a few of the more prominent langauges in the project. At the very least, you need a ```language:``` line with the project language in all lower case. Follow that with the language itself as a line, and the versions of your language you want to test on proceeding lines.
+
+That's the basics. For each language, however, Travis-CI uses a default mechanism for running the tests. For example, for Perl(see here: http://docs.travis-ci.com/user/languages/perl/), this first searches for a Build.PL file in the repo's root. If it finds one, it runs
+```
+perl Build.PL && .Build test
+```
+if not, it looks for a Makefile.PL, and runs it with
+```
+perl Makefile.PL && make test
+```
+if neither of those are found, it just runs
+```
+make test
+```
+
+Your .travis.yml file, however, is very configurable. In general, you'll probably use something similar to below - this is taken from the [Narrative Method Specs](https://github.com/kbase/narrative_method_specs) store. That store has a small script that just validates the format of each spec file.
+
+```
+language: perl
+
+perl:
+  - "5.18"
+
+install:
+  - cpanm --verbose --notest JSON::RPC::Client
+  - cpanm --verbose --notest Exception::Class
+  - cpanm --verbose --notest File::Slurp
+
+script: make test
+
+branches:
+  only:
+  - master
+  - staging
+  - develop
+  - dev
+```
+
+The test script is written in Perl, so it sets the language as Perl. The ```install:``` command runs each proceeding line in order before running anything. Note that this can include a ```git clone``` line to pull in any dependencies.
```

Comment:
Minor comment: when putting code blocks inline, a single pair of backticks should suffice, triple backticks are only needed for multiline blocks (as you used above). It keeps the document a bit more readable in source form.